File: Modul3/app/src/main/java/com/example/modul3/MainActivity.java
package com.example.modul3;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText emailInput = findViewById(R.id.emailInput);
        EditText passwordInput = findViewById(R.id.passwordInput);
        Button btnLogin = findViewById(R.id.btnLogin);

        btnLogin.setOnClickListener(v -> {
            String email = emailInput.getText().toString().trim();
            String password = passwordInput.getText().toString().trim();

            String correctEmail = "alfatadzaky@gmail.com";
            String correctPassword = "235150707111008";

            if (email.equals(correctEmail) && password.equals(correctPassword)) {
                Intent intent = new Intent(MainActivity.this, MrHeadActivity.class);
                intent.putExtra("email", email);
                intent.putExtra("password", password);
                startActivity(intent);
            } else {
                Toast.makeText(MainActivity.this, "Email atau Password salah!", Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: firebasepraktikum/app/src/main/java/com/example/firebasepraktikum/MainActivity.kt
package com.example.firebasepraktikum

import android.content.ContentValues
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var btnLoginAuto: Button? = null // tombol autentikasi otomatis
    private var mAuth: FirebaseAuth? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnLoginAuto = findViewById(R.id.btn_login_auto) // ambil referensi dari tombol baru
        mAuth = FirebaseAuth.getInstance()

        btnLoginAuto!!.setOnClickListener(this)
    }

    override fun onStart() {
        super.onStart()
        val currentUser = mAuth!!.currentUser
        updateUI(currentUser)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.btn_login_auto -> autoLogin() // tombol untuk login otomatis
        }
    }

    // Metode untuk autentikasi otomatis
    private fun autoLogin() {
        val email = "person1@gmail.com" // email yang sudah ditentukan
        val password = "person1" // password yang sudah ditentukan

        mAuth!!.signInWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    val user = mAuth!!.currentUser
                    updateUI(user)
                } else {
                    Toast.makeText(this@MainActivity, "Authentication failed.", Toast.LENGTH_SHORT).show()
                    updateUI(null)
                }
            }
    }

    private fun updateUI(user: FirebaseUser?) {
        if (user != null) {
            val intent = Intent(this@MainActivity, InsertNoteActivity::class.java)
            startActivity(intent)
        } else {
            Toast.makeText(this@MainActivity, "Log In First", Toast.LENGTH_SHORT).show()
        }
    }
}